csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -0,0 +1,954 @@
+            string catalog,
+            string dbSchemaPattern,
+            string tableNamePattern,
+            List<string> tableTypes,
+            string columnNamePattern)
+        {
+            StringArray.Builder dbSchemaNameBuilder = new StringArray.Builder();
+            List<IArrowArray?> dbSchemaTablesValues = new List<IArrowArray?>();
+            ArrowBuffer.BitmapBuilder nullBitmapBuffer = new ArrowBuffer.BitmapBuilder();
+            int length = 0;
+
+            string dbSchemaRegexp = PatternToRegEx(dbSchemaPattern);
+
+            PagedEnumerable<DatasetList, BigQueryDataset> schemas = this.client.ListDatasets(catalog);
+
+            foreach (BigQueryDataset schema in schemas)
+            {
+                if (Regex.IsMatch(schema.Reference.DatasetId, dbSchemaRegexp, RegexOptions.IgnoreCase))
+                {
+                    dbSchemaNameBuilder.Append(schema.Reference.DatasetId);
+                    length++;
+                    nullBitmapBuffer.Append(true);
+
+                    if (depth == GetObjectsDepth.DbSchemas)
+                    {
+                        dbSchemaTablesValues.Add(null);
+                    }
+                    else
+                    {
+                        dbSchemaTablesValues.Add(GetTableSchemas(
+                            depth, catalog, schema.Reference.DatasetId,
+                            tableNamePattern, tableTypes, columnNamePattern));
+                    }
+                }
+            }
+
+            List<IArrowArray> dataArrays = new List<IArrowArray>
+            {
+                dbSchemaNameBuilder.Build(),
+                CreateNestedListArray(dbSchemaTablesValues, new StructType(StandardSchemas.TableSchema)),
+            };
+
+            return new StructArray(
+                new StructType(StandardSchemas.DbSchemaSchema),
+                length,
+                dataArrays,
+                nullBitmapBuffer.Build());
+        }
+
+        private StructArray GetTableSchemas(
+            GetObjectsDepth depth,
+            string catalog,
+            string dbSchema,
+            string tableNamePattern,
+            List<string> tableTypes,
+            string columnNamePattern)
+        {
+            StringArray.Builder tableNameBuilder = new StringArray.Builder();
+            StringArray.Builder tableTypeBuilder = new StringArray.Builder();
+            List<IArrowArray?> tableColumnsValues = new List<IArrowArray?>();
+            List<IArrowArray?> tableConstraintsValues = new List<IArrowArray?>();
+            ArrowBuffer.BitmapBuilder nullBitmapBuffer = new ArrowBuffer.BitmapBuilder();
+            int length = 0;
+
+            string query = string.Format("SELECT * FROM `{0}`.`{1}`.INFORMATION_SCHEMA.TABLES",
+                catalog, dbSchema);
+
+            if (tableNamePattern != null)
+            {
+                query = string.Concat(query, string.Format(" WHERE table_name LIKE '{0}'", tableNamePattern));
```

Comment:
tableNamePattern comes from the caller, making this susceptible to SQL injection

Reply:
fixed in latest push